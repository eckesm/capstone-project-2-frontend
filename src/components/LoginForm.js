import React from 'react';
import { useDispatch } from 'react-redux';
import { useHistory } from 'react-router';

import useFields from '../hooks/useFields';
import useLocalStorageState from '../hooks/useLocalStorageState';

import { getAndStoreToken, getAndStoreUserInfo } from '../actions/auth';

export default function LoginForm() {
	const dispatch = useDispatch();
	const history = useHistory();

	const [ emailAddress, setEmailAddress ] = useLocalStorageState('email_address') || '';
	const initialState = {
		emailAddress : emailAddress || '',
		password     : ''
	};
	const [ formData, handleChange, resetFormData ] = useFields(initialState);

	async function handleSubmit(evt) {
		evt.preventDefault();
		try {
			const res = await dispatch(getAndStoreToken(formData.emailAddress, formData.password));
			if (res.status === 200) {
				setEmailAddress(formData.emailAddress);
				try {
					const userRes = await dispatch(getAndStoreUserInfo());
					if (userRes.status === 200) {
						history.push('/');
					}
				} catch (err) {
					console.log('handleSubmit() > getAndStoreToken() > getAndStoreUserInfo() error:', err);
				}
			}
		} catch (err) {
			console.log('handleSubmit() error:', err);
		}
	}

	return (
		<div>
			<h1>Login</h1>
			<form onSubmit={handleSubmit}>
				<div>
					<label htmlFor="emailAddress">Email Address:</label>
					<input
						type="text"
						id="emailAddress"
						value={formData.emailAddress}
						name="emailAddress"
						onChange={handleChange}
					/>
				</div>
				<div>
					<label htmlFor="password">Password:</label>
					<input
						type="password"
						id="password"
						value={formData.password}
						name="password"
						onChange={handleChange}
					/>
				</div>
				<button type="submit">Login</button>
			</form>
		</div>
	);
}
