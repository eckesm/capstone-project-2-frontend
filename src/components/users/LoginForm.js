import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { useHistory } from 'react-router';

import useFields from '../../hooks/useFields';
import useLocalStorageState from '../../hooks/useLocalStorageState';

import { getAndStoreUserInfo } from '../../actions/auth';
import { BackendApi } from '../../api/api';

import SubmitButton from '../buttons/SubmitButton';
import ErrorMessages from '../ErrorMessages';

import '../screen.css';
import './users.css';

export default function LoginForm() {
	const dispatch = useDispatch();
	const history = useHistory();

	const [ emailAddress, setEmailAddress ] = useLocalStorageState('email_address') || '';
	const initialState = {
		emailAddress : emailAddress || '',
		password     : ''
	};
	const [ formData, handleChange, resetFormData ] = useFields(initialState);

	const [ errors, setErrors ] = useState([]);

	async function handleSubmit(evt) {
		console.log('handleSubmit run');
		evt.preventDefault();
		try {
			const res = await BackendApi.getAndStoreTokenApi(formData.emailAddress, formData.password);
			console.log('getStoreToken', res);
			if (res.status === 200) {
				setEmailAddress(formData.emailAddress);
				try {
					const userRes = await dispatch(getAndStoreUserInfo());
					if (userRes.status === 200) {
						history.push('/restaurants');
					}
				} catch (err) {
					console.log('handleSubmit() > getAndStoreToken() > getAndStoreUserInfo() error:', err);
				}
			}
			if (res.status === 401) {
				setErrors([ res.message ]);
			}
		} catch (err) {
			console.log('handleSubmit() error:', err);
		}
	}

	return (
		<form className="LoginForm BasicView" onSubmit={handleSubmit}>
			<div className="Section">
				<div className="InputGroup">
					<label htmlFor="emailAddress">Email Address:</label>
					<input
						type="text"
						id="emailAddress"
						value={formData.emailAddress}
						name="emailAddress"
						onChange={handleChange}
						required
					/>
				</div>
				<div className="InputGroup">
					<label htmlFor="password">Password:</label>
					<input
						type="password"
						id="password"
						value={formData.password}
						name="password"
						onChange={handleChange}
						required
					/>
				</div>
			</div>
			<SubmitButton text="Login" />
			<ErrorMessages errors={errors} />
		</form>
	);
}
