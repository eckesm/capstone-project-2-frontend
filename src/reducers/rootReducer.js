import { combineReducers } from 'redux';
import active from './active';
import auth from './auth';
// import budget from './budget';
import invoices from './invoices';
import restaurants from './restaurants';
import sales from './sales';

const rootReducer = combineReducers({ active, auth, invoices, restaurants, sales });
// const rootReducer = combineReducers({ active, auth, budget, invoices, restaurants, sales });

export default rootReducer;
