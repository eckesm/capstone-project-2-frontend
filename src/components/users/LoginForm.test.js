import React from 'react';
import { render } from '@testing-library/react';

import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from '../../reducers/rootReducer';
import { Provider } from 'react-redux';
import { HashRouter } from 'react-router-dom';
import { composeWithDevTools } from 'redux-devtools-extension';

import LoginForm from './LoginForm';
import { MemoryRouter } from 'react-router';

it('matches snapshot', function() {
    const store = createStore(rootReducer, composeWithDevTools(applyMiddleware(thunk)));
	const { asFragment } = render(
		<Provider store={store}>
			<MemoryRouter>
				<LoginForm />
			</MemoryRouter>
		</Provider>
	);
	expect(asFragment()).toMatchSnapshot();
});
