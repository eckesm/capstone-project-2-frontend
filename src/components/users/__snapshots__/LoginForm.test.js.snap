// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="LoginForm BasicView"
  >
    <div
      class="Section"
    >
      <div
        class="InputGroup"
      >
        <label
          for="emailAddress"
        >
          Email Address:
        </label>
        <input
          id="emailAddress"
          name="emailAddress"
          required=""
          type="text"
          value=""
        />
      </div>
      <div
        class="InputGroup"
      >
        <label
          for="password"
        >
          Password:
        </label>
        <input
          id="password"
          name="password"
          required=""
          type="password"
          value=""
        />
      </div>
    </div>
    <button
      class="CustomButton"
      type="submit"
    >
      Login
    </button>
    <ul
      class="IgnoreList"
    />
  </form>
</DocumentFragment>
`;
